tests/integration.rs

```diff
@@ -5664,6 +5669,46 @@ mod parallel {
         handle_child_output(r, &output);
     }
 
+    #[test]
+    #[cfg(feature = "guest_debug")]
+    fn test_coredump() {
+        let focal = UbuntuDiskConfig::new(FOCAL_IMAGE_NAME.to_string());
+        let guest = Guest::new(Box::new(focal));
+        let api_socket = temp_api_path(&guest.tmp_dir);
+
+        let mut cmd = GuestCommand::new(&guest);
+        cmd.args(&["--cpus", "boot=1"])
+            .args(&["--memory", "size=512M"])
+            .args(&["--kernel", fw_path(FwType::RustHypervisorFirmware).as_str()])
+            .default_disks()
+            .args(&["--net", guest.default_net_string().as_str()])
+            .args(&["--api-socket", &api_socket])
+            .capture_output();
+
+        let mut child = cmd.spawn().unwrap();
+        let vmcore_file = temp_vmcore_file_path(&guest.tmp_dir);
+
+        let r = std::panic::catch_unwind(|| {
+            guest.wait_vm_boot(None).unwrap();
+
+            assert!(remote_command(&api_socket, "pause", None));
+
+            assert!(remote_command(
+                &api_socket,
+                "coredump",
+                Some(format!("file://{}", vmcore_file).as_str()),
+            ));
```

Comment:
That's a first step to check if `ch-remote` returns an error, but I also think we could go one step further and maybe check the header of the generated file, along with the file size which we should know.
@rbradford wdyt?